lib/rspec/matchers/built_in/respond_to.rb

```diff
@@ -125,7 +125,14 @@ def matches_arity?(actual, name)
 
           return true if expectation.empty?
 
-          signature = Support::MethodSignature.new(Support.method_handle_for(actual, name))
+          method_handle = Support.method_handle_for(actual, name)
+
+          signature =
+            if name == :new && method_handle.owner === ::Class && Class === actual
+              Support::MethodSignature.new(actual.instance_method(:initialize))
+            else
+              Support::MethodSignature.new(method_handle)
+            end
```

Comment:
Nice to see how simple this is!

Reply:
I'm wondering if we need to do the same hoops for an instance_method as a method